packages/babel-plugin-minify-dead-code-elimination/src/index.js

```diff
@@ -228,21 +228,29 @@ module.exports = ({ types: t, traverse }) => {
               continue;
             }
 
-            if (
-              binding.path.isVariableDeclarator() &&
-              binding.path.node.init &&
-              !scope.isPure(binding.path.node.init) &&
-              binding.path.parentPath.node.declarations
-            ) {
-              if (binding.path.parentPath.node.declarations.length !== 1) {
-                continue;
-              }
-              // Bail out for ArrayPattern and ObjectPattern
+            if (binding.path.isVariableDeclarator()) {
               if (!binding.path.get("id").isIdentifier()) {
+                // deopt for object and array pattern
                 continue;
               }
 
-              binding.path.parentPath.replaceWith(binding.path.node.init);
+              // if declarator has some impure init expression
+              // var x = foo();
+              // => foo();
+              if (
+                binding.path.node.init &&
+                !scope.isPure(binding.path.node.init) &&
+                binding.path.parentPath.node.declarations
```

Comment:
I'm currently getting the error `Cannot read property 'declarations' of null` here. Doing steps to reproduce is a lot of work (big repo). So I thought I would take a wild stab in the dark and ask if you could see that it would in any way make sense to add a guard here for if `binding.path.parentPath.node` is null? If that doesn't make sense I apologise for wasting your time...